.build/CodeFormatter.ps1

```diff
@@ -28,9 +31,63 @@ if (-not (Load-Module -Name EncodingAnalyzer)) {
 
 $repoRoot = Get-Item "$PSScriptRoot\.."
 
-$filesToCheck = Get-ChildItem -Path $repoRoot -Directory | Where-Object {
-    $_.Name -ne "dist" } | ForEach-Object {
-    Get-ChildItem -Path $_.FullName -Include "*.ps1", "*.psm1", "*.md" -Recurse
+$optimizeCodeFormatter = $AllFiles -eq $false
+$filesFullPath = New-Object 'System.Collections.Generic.List[string]'
+# Get only the files that are changed in this PR
+if ($optimizeCodeFormatter) {
+
+    Write-Verbose "Checking commits only"
+    # Get all the commits between origin/main and HEAD.
+    $gitlog = git log --format="%H %cd" --date=rfc origin/main..HEAD
+    $m = $gitlog | Select-String "^(\S+) (.*)$"
+
+    foreach ($commitMatch in $m) {
+        $commitHash = $commitMatch.Matches.Groups[1].Value
+        $filesChangedInCommit = git diff-tree --no-commit-id --name-only -r $commitHash
+
+        foreach ($fileChanged in $filesChangedInCommit) {
+            $fullPath = Join-Path $repoRoot $fileChanged
+
+            if ($filesFullPath.Contains($fullPath)) {
```

Comment:
Same here. HashSet makes this check faster in the case of a large number of changed files. That's why ValidateMerge uses a HashSet here.